source/comp/markv_codec.cpp

```diff
@@ -152,6 +152,20 @@ const size_t kCommentNumWhitespaces = 2;
 
 const size_t kByteBreakAfterInstIfLessThanUntilNextByte = 8;
 
+const uint32_t kShortDescriptorNumBits = 8;
+
+// Custom hash function used to produce short descriptors.
+uint32_t ShortHashU32Array(const std::vector<uint32_t>& words) {
+  // The hash function is a sum of hashes of each word seeded by word index.
+  // Knuth's multiplicative hash is used to hash the words.
+  const uint32_t kKnuthMulHash = 2654435761;
+  uint32_t val = 0;
+  for (uint32_t i = 0; i < words.size(); ++i) {
+    val += (words[i] + i + 123) * kKnuthMulHash;
+  }
+  return 1 + val % ((1 << kShortDescriptorNumBits) - 1);
```

Comment:
I wonder if the % is slow.  But performance measurement would show that. Never mind.